### PR TITLE
Add Thread.GetCurrentProcessorId() API

### PR DESCRIPTION
**Pull request by jkotas**
Contributes to https://github.com/dotnet/corefx/issues/16767

### CODE REVIEW

File: src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs [hunk 1]
```diff
@@ -213,6 +213,49 @@ internal static int OptimalMaxSpinWaitsPerSpinIteration
...
+        {
+            int currentProcessorId = GetCurrentProcessorNumber();
+
+            // On Unix, GetCurrentProcessorNumber() is implemented in terms of sched_getcpu, which
+            // doesn't exist on all platforms.  On those it doesn't exist on, GetCurrentProcessorNumber()
+            // returns -1.  As a fallback in that case and to spread the threads across the buckets
+            // by default, we use the current managed thread ID as a proxy.
+            if (currentProcessorId < 0) currentProcessorId = Environment.CurrentManagedThreadId;
+
+            Debug.Assert(ProcessorIdRefreshRate <= ProcessorIdCacheCountDownMask);
```
Comment (ahsonkhan) [inline src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs]: Why we do we need this debug assert for comparing two const values?

File: src/vm/ecalllist.h [hunk 2]
```diff
@@ -690,6 +689,7 @@ FCFuncStart(gRuntimeThreadFuncs)
     FCFuncElement("InterruptInternal", ThreadNative::Interrupt)
     FCFuncElement("JoinInternal", ThreadNative::Join)
     QCFuncElement("GetOptimalMaxSpinWaitsPerSpinIterationInternal", ThreadNative::GetOptimalMaxSpinWaitsPerSpinIteration)
+    FCFuncElement("GetCurrentProcessorNumber", ThreadNative::GetCurrentProcessorNumber)
```
Comment (ahsonkhan) [inline src/vm/ecalllist.h]: rename to GetCurrentProcessorNumberInternal?

File: src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs (hunk 1 above)
Comment (stephentoub) [reply src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs]: Sort of the equivalent of a static_assert in C++; you want to validate that your consts and things computed at compile time make sense.

File: src/vm/ecalllist.h (hunk 2 above)
Comment (jkotas) [reply src/vm/ecalllist.h]: FCall naming conventions say to use the Internal prefix only if necessary. Internal is not necessary here. https://github.com/dotnet/coreclr/blob/master/Documentation/botr/mscorlib.md#naming-convention.

**Pull request merged**